plugins/module_utils/network/sonic/config/bgp_af/bgp_af.py

```diff
@@ -999,3 +1085,190 @@ def get_delete_bgp_af_requests(self, commands, have, is_delete_all):
                 match_cfg = next((have_cfg for have_cfg in have if have_cfg['vrf_name'] == vrf_name and have_cfg['bgp_as'] == as_val), None)
             requests.extend(self.get_delete_single_bgp_af_request(cmd, is_delete_all, match_cfg))
         return requests
+
+    def get_delete_commands_requests_for_replaced_overridden(self, want, have, state):
```

Comment:
There's 'get_replaced_config' in utils that's used in a few other modules are you able to make use of it? Perhaps it will simplify things.

Reply:
The output of 'get_replaced_config' in utils cannot be used for deletion in 'bgp_af' due to its following characteristics:
- Whether the entire 'address_family' or 'vni' is to be deleted, or only specific attributes in them needs to be deleted cannot be determined directly from it.
- When one of the trivial 'address_family' attributes not in the key_set such as 'dampening' is modified or not present in the new config, the result returned contains the entire existing config of that address_family. This results in deleting the entire config and then configuring the entire new config, instead of deleting/modifing that particular attribute alone.

Reply:
Thanks for explaining.